pkg/catalog/helpers_test.go

```diff
@@ -24,7 +24,11 @@ import (
 	"github.com/openservicemesh/osm/pkg/tests"
 )
 
-func newFakeMeshCatalogForRoutes(t *testing.T) *MeshCatalog {
+type testParams struct {
+	permissiveMode bool
+}
+
+func newFakeMeshCatalogForRoutes(t *testing.T, testParams testParams) *MeshCatalog {
```

Comment:
In the future, lets remove these additional fakes and rely on building catalog using mocks of `smi.MeshSpec` and `kuberenetes.Controller`, or this function is going to bloat up and new fakes will be required based on the test variability.

This also makes it very clear in the test what the test is expecting and why, instead of the magic happening inside these fake functions.

Reply:
I can definitely agree this may be brittle in the long run. Let's discuss this further bc the fake method is indeed creating and setting mocks. Do you have an example of the kind of set up you think is ideal?

Reply:
The fake is indeed creating mocks, but it is also relying on static fake SMI and Kube clients which are not extendable for tests. The problem with using static fake clients (smi, kubernetes) in table driven tests is that the APIs will return the same objects (ex. services, traffic targets etc.) every time. Also the fakes of these interfaces has proven to be very problematic especially when coupled with mocks. For ex. refer to the tests that use a combination of fakes + mocks in the catalog pkg.